src/targets/maven.ts

```diff
@@ -0,0 +1,352 @@
+          'See the documentation for more details.'
+      );
+    }
+    return {
+      android: {
+        distDirRegex: this.config.android.distDirRegex,
+        fileReplaceeRegex: this.config.android.fileReplaceeRegex,
+        fileReplacerStr: this.config.android.fileReplacerStr,
+      },
+    };
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   * If there's required software missing, raises an error.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug(
+      'Checking if Maven CLI is available: ',
+      this.mavenConfig.mavenCliPath
+    );
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
+    this.logger.debug(
+      'Checking if Gradle CLI is available: ',
+      this.mavenConfig.gradleCliPath
+    );
+    checkExecutableIsPresent(this.mavenConfig.gradleCliPath);
+    this.logger.debug('Checking if GPG is available in the path...');
+    checkExecutableIsPresent('gpg');
+  }
+
+  /**
+   * Parses the required target's configuration variables' types from strings to
+   * RegExp.
+   */
+  private makeRegexpFromConfiguration(): void {
+    this.config.android.distDirRegex = stringToRegexp(
+      this.config.android.distDirRegex
+    );
+    this.config.android.distDirRegex = stringToRegexp(
+      this.config.android.fileReplaceeRegex
+    );
+  }
+
+  /**
+   * Publishes current Java and Android distributions.
+   * @param version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revison: string): Promise<void> {
+    await this.createUserGradlePropsFile();
+    await this.upload(revison);
+    await this.closeAndRelease();
+  }
+
+  /**
+   * Creates the required user's `gradle.properties` file.
+   *
+   * If there's an existing one, it's overwritten.
+   * TODO: control when it's overwritten with an option.
+   */
+  public createUserGradlePropsFile(): Promise<void> {
+    return fsPromises.writeFile(
+      join(this.getGradleHomeDir(), GRADLE_PROPERTIES_FILENAME),
+      [
+        'mavenCentralUsername=' + this.mavenConfig.MAVEN_CENTRAL_USERNAME,
+        'mavenCentralPassword=' + this.mavenConfig.MAVEN_CENTRAL_PASSWORD,
+      ].join('\n')
+    );
+  }
+
+  /**
+   * Retrieves the Gradle Home path.
+   *
+   * @returns the gradle home path.
+   */
+  public getGradleHomeDir(): string {
+    if (process.env.GRADLE_USER_HOME) {
+      return process.env.GRADLE_USER_HOME;
+    }
+
+    return DEFAULT_GRADLE_USER_HOME;
+  }
+
+  /**
+   * Uploads the artifacts with the required files. This is a required step
+   * to make a release, but this doesn't perform any releases; after upload,
+   * the flow must finish with `closeAndRelease`.
+   */
+  public async upload(revision: string): Promise<void> {
+    this.logger.debug('Fetching artifact list...');
```

Comment:
No need for this as we already have a more detailed debug log from the base function: https://github.com/getsentry/craft/blob/150310a11c4896e86df8ca26ef36e10ca5a44c8d/src/targets/base.ts#L85
```suggestion
```